Applications/Spire/Include/Spire/Ui/ColorPicker.hpp

```diff
@@ -0,0 +1,70 @@
+#ifndef SPIRE_COLOR_PICKER_HPP
+#define SPIRE_COLOR_PICKER_HPP
+#include <memory>
+#include <QColor>
+#include <QWidget>
+#include "Spire/Spire/ListModel.hpp"
+#include "Spire/Spire/ValueModel.hpp"
+#include "Spire/Ui/Ui.hpp"
+
+namespace Spire {
+
+  /**
+   * Displays a panel where the user can browse and select a color.
+   */
+  class ColorPicker : public QWidget {
+    public:
+
+      /**
+       * Constructs a ColorPicker with a default local model and
+       * a predefined set of colors.
+       * @param parent The parent widget.
+       */
+      explicit ColorPicker(QWidget& parent);
+
+      /**
+       * Constructs a ColorPicker with a predefined set of colors.
+       * @param current The model used for the current color.
+       * @param parent The parent widget.
+       */
+      ColorPicker(std::shared_ptr<ValueModel<QColor>> current,
```

Comment:
Are there typedefs for these models?

    ColorValueModel
    LocalColorValueModel
    ...

Reply:
There are no these typedefs here. These typedefs were added to the `ColorBox.hpp`, and `ColorBox` are the parent of `ColorPicker`. Maybe we can put these typedefs to a separate file, like the typedefs for decimal models in `Decimal.hpp`. In this case, the `ColorBox`, `ColorPicker`, `ColorCodePanel` and even `HexColorBox` can use these typedefs.

Reply:
Feel free to include ```ColorBox.hpp```

Reply:
`ColorBox.hpp` does not exist yet. The ColorBox is a task I am currently implementing and has not yet been submitted.

Reply:
Understood, in that case this is fine. When you implement ```ColorBox```, replace all references to ```ValueModel<QColor>``` with the typedef. For this task it's fine as is.